src/main/java/gregtech/common/items/behaviors/DoorBehavior.java

```diff
@@ -0,0 +1,50 @@
+package gregtech.common.items.behaviors;
+
+import gregtech.api.items.metaitem.stats.IItemBehaviour;
+import net.minecraft.block.Block;
+import net.minecraft.block.SoundType;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemDoor;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.*;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+public class DoorBehavior implements IItemBehaviour {
+
+    private final Block block;
+
+    public DoorBehavior(Block block) {
+        this.block = block;
+    }
+
+    @Override
+    public ActionResult<ItemStack> onItemUse(EntityPlayer player, World world, BlockPos pos, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ) {
+        // Copied from ItemDoor
```

Comment:
Why can't ItemDoor be used directly?

Reply:
ItemDoor is not an itemblock, so it's a little bit different than regular blocks. I just added metaitems instead of adding new item to registry because we already have implementation for metaitems.